[docs] update expo upload docs (#8425)
expo/expo: docs/pages/distribution/uploading-apps.md

To upload the previously built standalone app to the appropriate app store, you simply run `expo upload:android` or `expo upload:ios`. However, you have a few options for choosing which app binary you want to upload (remember to choose one at the time):

- `--latest` - uploads the latest build for the given platform found on Expo Servers
- `--url <url>` - uploads a build from given URL
- `--path <path>` - uploads a build from the local file system
- `--id <id>` - uploads a build with the given ID

## 2.1. If you choose to upload your Android app to Google Play

**Important:**

> - Beware that this feature works properly only on macOS, unless you pass the `--use-submission-service` flag.

Reviewer: ```suggestion
- Beware that this feature works properly only on macOS unless you pass the `--use-submission-service` flag.
```

After these steps, you can make use of `expo-cli` to upload your further app builds to Google Play.

To upload your Android app to Google Play, simply run `expo upload:android`. You can set following options when uploading an Android standalone app:

- `--type <archive-type>` - the archive type, by default, it's inferred from the filename extension, choose from: apk, aab
- `--key <key>` - path to the JSON key used to authenticate with the Google Play Store
- `--track <track>` - the track of the application to use, choose from: production, beta, alpha, internal, rollout (default: internal)
- `--release-status <release-status>` - release status (used when uploading new apks/aabs), choose from: completed, draft, halted, inProgress (default: completed)
- `--android-package <android-package>` - the Android package of your app, if you don't pass this parameter it'll be read from `app.json`
- `--use-submission-service` - **Experimental:** Use Submission Service for uploading your app. The upload process will happen on Expo servers.
- `--verbose` - always print logs from Submission Service

#### Uploading your app with Submission Service

> **Beware:** This feature is still experimental!

Reviewer: we should clarify this to indicate that this won't cause any damage, worst case is that their submission just may not work

> **Beware:** This feature is still experimental! However, using it can not cause any damage to your app. In the worst-case scenario, your app won't get submitted to Google Play Store.

Author: @brentvatne how about now?

> If you'd like to upload your app from any other operating system than macOS, you can give Submission Service a try. Submission Service is a part of Expo Application Services. It lets you upload your standalone app directly from Expo Servers.

Reviewer: ```suggestion
If you would like to upload your app from any other operating system than macOS, you can give the Expo Submission Service a try. It lets you upload your standalone app directly from Expo Servers.
```

Reviewer: i only removed Expo Application Services here because we haven't used the term anywhere else on our website yet and so it could confuse people if we start to use it before we have some other marketing material or docs for it

> Using Submission Service is really easy - you just need to pass an additional flag: `expo upload:android --use-submission-service`.

Reviewer: ```suggestion
Using the Submission Service is easy - it's just one additional flag: `expo upload:android --use-submission-service`.
```

## 2.2. If you choose to upload your iOS app to TestFlight

> **Disclaimer:** This feature works properly only on macOS.

Reviewer: ```suggestion
**Disclaimer:** This feature only works on macOS. If you want to upload an iOS app to the Apple App Store you will need to have access to a macOS device.
```
